pandas/core/arrays/string_arrow.py

```diff
@@ -858,6 +859,14 @@ def _str_isupper(self):
         else:
             return super()._str_isupper()
 
+    def _str_len(self):
+        # utf8_length added in pyarrow 4.0.0
```

Comment:
we should do this as a more general check i think (e.g. maybe defined a pyarrow_gt_400 or similar to what we do elsewhere rather than rely on checks like this (e.g. create a pyarrow.compat or similar). but let's create an issue and do a followup.

Reply:
sgtm